### DISCUSSION TITLE
CountFragments Output dataframe question

### DISCUSSION THREAD

**Discussion by gandreeva18** [stuart-lab/signac#1471] [Q&A] [answered]
Good afternoon, 

I have seen that I can run the Signac pipeline using only a fragments file and its index ([#1025](https://github.com/stuart-lab/signac/discussions/1025) and [#908 ](https://github.com/stuart-lab/signac/discussions/908)). 

When I run CountFragments, I obtain a dataframe with many more cell barcodes than expected. The problem is that, the authors stated no more than 10,000 nuclei per sample were processed for scATACseq. 

`total_fragments <- CountFragments("GSM5952255_E775_1_fragments.tsv.gz") `
`length(unique(total_fragments$CB))` 
the output is 377223

The data is from: [GEO accession](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM5952255 )

Am I doing or understanding something wrong?

Thank you so much for your help :) 
GS

**Comment by timoast** [comment, marked as answer]
The `CountFragments` function will return the counts for each barcode in the fragment file, unless you set the `cells` parameter. It is likely that the fragment file you have contains some fragments for barcodes that were not used in the final analysis.
